Update code to match the project style guide

settermjd opened the pull request:
Please check the type of change your PR introduces.

* [ ] Bugfix
* [ ] Documentation
* [ ] New Feature
* [ ] RFC
* [X] QA
* [ ] Other
* [ ] BC Break?

Description:

 This change cleans up the code to ensure that it passes all QA tests.

settermjd committed:
Update code to match the project style guide

 This change is a result of running phpcbf over the code, and performing
 some manual changes to ensure that the code passes the phpcs
 configuration.

settermjd committed:
Only pass a string to strtolower

As pointed out by Psalm, strtolower only accepts a string, but
getAttribute returns mixed. So, in the most simplistic way I know, this
change refactors to the code to ensure that the driver name is only
passed from getAttribute to strtolower if it is a string.

settermjd committed:
Annotate Module as being part of the public API

This change has been made as Psalm can't detect usage of the class on
its own, so thinks that it is dead code, when it likely isn't; at least
not at the moment.

settermjd committed:
Properly annotate the structure of $config

settermjd committed:
Pass all required arguments to initialisation of Pdo

Pdo's constructor requires three mandatory arguments: a
AbstractPdoConnection|PDO object, a
StatementInterface&PdoDriverAwareInterface object, and a ResultInterface
object. This change mocks the second and third arguments to ensure,
which were missing.

settermjd committed:
Suppress a Psalm InvalidArgument failure

I don't see this Psalm failure as an issue for the time being, so am
ignoring it in this change, quite visibly so that it will be hard to
miss, and attended to in the future.

tyrsson commented on src/Platform/Sqlite.php:
```diff
@@ -43,7 +42,7 @@ public function quoteValue(string $value): string
         }
 
-        if ($resource instanceof \PDO) {
+        if ($resource instanceof PDO) {
```
When this is changed the import for *\Pdo\Pdo has to change with it or it causes issues for devs in windows.

settermjd replied:
I'll check in to that. Thanks for pointing it out.

settermjd replied:
How does it cause issues for Windows developers?

tyrsson replied:
It's due to the way windows handles upper/lower case. In windows there is no difference between PDO and Pdo.

tyrsson replied:
I can handle the \PDO to PDO changes. It will be easier for me since I dev on windows. The code will still run (in most cases) but the DX will be extremely degraded since one or the other will show as a "class not found" or whatever their particular ide displays for that. Plus, if there is a typo in anything that is already showing as a false positive its really really hard to find since the ide will not show it to you.

I'll review it here shortly.

settermjd committed:
Update the Psalm baseline

The change is required to reflect the changes in the package, or to
properly reflect the package.

settermjd committed:
Update the code following static analysis review

This change is a minor one, changing the code, improving it based on the
results of the review.

settermjd committed:
Minor code tightening

This is a small change aimed at tightening up the test code. Removing
items that are unnecessary, such as checking for the pdo_sqlite
extension, and adding in required function arguments, to ensure that the
code is as explicit as possible.

tyrsson reviewed (approved):
Approved.

tyrsson commented:
Why is PHPCS still failing?

tyrsson commented:
composer platform needs updated to 8.2.99
